crates/ruff_cli/src/lib.rs

```diff
@@ -210,16 +246,19 @@ fn check(args: CheckArgs, log_level: LogLevel) -> Result<ExitStatus> {
             watcher.watch(file, RecursiveMode::Recursive)?;
         }
 
+        let mut file_change_detector = FileChangeDetector::new();
         loop {
             match rx.recv() {
                 Ok(event) => {
-                    let paths = event?.paths;
-                    let py_changed = paths.iter().any(|path| {
-                        path.extension()
-                            .map(|ext| ext == "py" || ext == "pyi")
-                            .unwrap_or_default()
-                    });
-                    if py_changed {
+                    if file_change_detector.change_detected(&event?.paths) {
+                        if file_change_detector.has_configuration_changed {
+                            pyproject_strategy = resolve::resolve(
+                                cli.isolated,
+                                cli.config.as_deref(),
+                                &overrides,
+                                cli.stdin_filename.as_deref(),
+                            )?;
```

Comment:
Only re-computing the `pyproject_strategy`, is, unfortunately, not sufficient. There are multiple derived variables on line 107-185 that need to be recomputed too

Reply:
I added a TODO for this.